src/HDF5Msg.chpl

```diff
@@ -1510,6 +1881,98 @@ module HDF5Msg {
         return (dset, "seg_string", "%s+%t".format(stringsEntry.name, stringsEntry.nBytes));
     }
 
+    proc segarray_readhdfMsg(filenames: [?fD] string, dset: string, dataclass, bytesize: int, isSigned: bool, validFiles: [] bool, st: borrowed SymTab): (string, string, string) throws {
+        h5Logger.debug(getModuleName(),getRoutineName(),getLineNumber(),
+            "Filename: %jt, dset: %s, dataclass: %i, bytesize: %i, isSigned: %i, validFiles: %jt".format(
+                filenames,
+                dset,
+                dataclass,
+                bytesize,
+                isSigned,
+                validFiles
+            )
+        );
+        
+        var subdoms: [fD] domain(1);
+        var segSubdoms: [fD] domain(1);
+        var skips = new set(string);
+        var len: int;
+        var nSeg: int;
+
+        (segSubdoms, nSeg, skips) = get_subdoms(filenames, dset + "/" + SEGMENTED_OFFSET_NAME, validFiles);
+        (subdoms, len, skips) = get_subdoms(filenames, dset + "/" + SEGMENTED_VALUE_NAME, validFiles);
+
+        var segEntry = new shared SymEntry(nSeg, int);
```

Comment:
I would make the dist array and then pass that into `read_files_into_distributed_array` because this should be more performant.
```
var segDist = makeDistArray(nSeg, int);
```

Reply:
Updated